components/base/kinematicbase/ptgKinematics.go

```diff
@@ -0,0 +1,122 @@
+// Package kinematicbase contains wrappers that augment bases with information needed for higher level
+// control over the base
+package kinematicbase
+
+import (
+	"context"
+	"errors"
+	"math"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"go.uber.org/multierr"
+	utils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/referenceframe"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+// Define a default speed to target for the base in the case where one is not provided.
+const defaultBaseMps = 0.6
+
+const (
+	ptgIndex int = iota
+	trajectoryIndexWithinPTG
+	distanceAlongTrajectoryIndex
+)
+
+type ptgBaseKinematics struct {
+	base.Base
+	frame referenceframe.Frame
+	fs    referenceframe.FrameSystem
+}
+
+// wrapWithPTGKinematics takes a Base component and adds a PTG kinematic model so that it can be controlled.
+func wrapWithPTGKinematics(
+	ctx context.Context,
+	b base.Base,
+) (KinematicBase, error) {
+	properties, err := b.Properties(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	if properties.TurningRadiusMeters == 0 {
+		return nil, errors.New("can only wrap with PTG kinematics if Base property TurningRadiusMeters is nonzero")
+	}
+
+	baseMetersPerSecond := defaultBaseMps // Currently no way to get this out of properties
```

Comment:
what if `defaultBaseMps` is not accurate? Should base speed be added to properties?
Does knowing the bases' speed make a difference for the planner?

Reply:
Since the planner is outputting velocities to be followed for lengths of time, it is necessary that it knows how fast the base is moving.

Soon we will be able to get this number out of the base and this will not be needed any longer.

But if the given speed is not accurate, the base will not follow the planned path accurately.